tools/codegen/gen_functionalization_type.py

```diff
@@ -0,0 +1,344 @@
+from tools.codegen.api import cpp
+from tools.codegen.api import dispatcher
+from tools.codegen.api.types import DispatcherSignature, CppSignatureGroup
+from tools.codegen.api.translate import translate
+from tools.codegen.context import method_with_native_function
+from tools.codegen.model import (
+    Argument, NativeFunction, NativeFunctionsGroup, Return, SchemaKind, BackendIndex,
+    DispatchKey, Variant, Tag, BaseType, BaseTy
+)
+from typing import List, Optional, Union, Tuple, Dict
+from dataclasses import dataclass
+from typing_extensions import Literal
+from tools.codegen.utils import mapMaybe, Target, assert_never
+
+from tools.autograd.gen_inplace_or_view_type import (
+    gen_formals, modifies_arguments
+)
+from tools.autograd.gen_trace_type import (
+    type_wrapper_name
```

Comment:
This also seems like a faux amis. I don't see why the wrapper names here have anything to do with the trace type type wrapper names.

Reply:
Agreed, it would probably be more sensible to use `dispatcher.name(f)` for the functions in the `at::functionalize` namespace (which separates any connection with tracing codegen, and also gives every op a name matching the C++ API function names, which is probably easier to remember).